cu_servicenow/cu_servicenow.module

```diff
@@ -0,0 +1,316 @@
+  if (isset($form_state['values']['servicenow']['field_map'])) {
+    $sn_value = [$form_state['values']['form_key'] => $form_state['values']['servicenow']['field_map']];
+    $sn_value = array_merge($current_vars, $sn_value);
+    variable_set('servicenow_field_mapping', $sn_value);
+    drupal_set_message(t('Successfully mapped webform field to ServiceNow.'), 'notice');
+  }
+}
+
+/**
+ * Custom submit handler for forms with SN integaration enabled.
+ */
+function servicenow_submit(&$form, &$form_state) {
+  global $user;
+  $nid = $form['#node']->nid;
+
+  // If the current node has SN enabled, do code, else move on.
+  watchdog('cu_servicenow', 'ServiceNow custom submit handler was hit.');
+  $form_values = $form_state['input']['submitted'];
+  $servicenow_field_map = variable_get('servicenow_field_mapping', '');
+
+  foreach($form_values as $key => $value) {
+    if(key_exists($key, $servicenow_field_map)) {
+      $sn[$servicenow_field_map[$key]] = $value;
+    }
+  }
+
+  $sn_incident_map = variable_get('sn_incident_' . $nid, []);
+
+  $sn['task_for'] = $user->name;
+  $sn['u_alternate_email'] = $user->mail;
+  $sn['state'] = '9';
+  $sn['contact_type'] = 'self-service';
+  $sn['cmdb_ci'] = $sn_incident_map['business_service']; // Business Service
+  $sn['u_nature_of_issue'] = $sn_incident_map['nature_of_issue']; // Nature of Issue
+  $sn['impact'] = '9c6238b555beb00096fffc196a1bb892'; // Impact
+  $sn['urgency'] = '9c6238b555beb00096fffc196a1bb892'; // Urgency
+
+  // @todo add incident assignments from form options/settings.
+
+  _servicenow_curl_request($sn);
+  //dpm($sn);
+
+  // Test uploading files
+  //_servicenow_upload_file($_FILES['files']['tmp_name']['submitted_attachment'], $_FILES['files']['type']['submitted_attachment'], $_FILES['files']['name']['submitted_attachment']);
+}
+
+function _servicenow_incident_assignments() {
+  global $conf;
+  $incident_assignment = variable_get('sn_environment', 'https://coloradosandbox.service-now.com') . '/api/now/v2/table/u_inc_assignment?sysparm_fields=sys_id,u_business_service,u_nature_of_issue,u_active,sys_updated_on&sysparm_display_value=true&sysparm_exclude_reference_link=true';
+
+  $ch = curl_init(); // initialize curl handle
+  curl_setopt($ch, CURLOPT_HTTPHEADER, Array("Accept:application/json","Content-Type:application/json"));
+  curl_setopt($ch, CURLOPT_USERPWD, $conf['sn_key']);
+  curl_setopt($ch, CURLOPT_URL, $incident_assignment);
+  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+  curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+  curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+  $result = curl_exec($ch); // run the whole process
+  curl_setopt($ch, CURLOPT_HEADER, 1);
+  curl_setopt($ch, CURLOPT_VERBOSE, 1);
+  //$header = curl_exec($ch); // run the whole process
+  curl_close($ch); // close cURL handler
+
+  //$headers = _get_headers_from_curl_response($header);
+  $result = json_decode($result);
+
+  return $result;
```

Comment:
The two calls to this function expect it to be an object with an array in $result->result. I didn't see any error checking on those calls so that needs to be addressed somehow.